xformers/benchmarks/benchmark_blocksparse_transformers.py

```diff
@@ -0,0 +1,1042 @@
+import gc
+import math
+from collections import namedtuple
+from dataclasses import dataclass
+from typing import Any, Dict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+import triton
+from triton.ops.blocksparse import matmul as blocksparse_matmul
+from yaml import BlockSequenceStartToken
+
+from xformers.components.attention.attention_patterns import (
+    axial_2d_pattern,
+    causal_1d_pattern,
+    global_token_pattern,
+    local_1d_pattern,
+    local_2d_pattern,
+)
+from xformers.components.attention.core import SparseCS, _matmul_with_mask
+
+device = "cuda"
+TestCase = namedtuple("TestCase", ["prepare_callable", "mask", "config", "name"])
+
+##############################################
+# Plotting utilities
+##############################################
+
+
+def pretty_barplot(results, title, units: str, filename=None, dash_key=""):
```

Comment:
nit but we had some utils (not as nice but these could be a complement) [here](https://github.com/facebookresearch/xformers/blob/main/xformers/benchmarks/utils.py). Maybe better to move this there for easier code discovery ?